stylix/mk-target.nix

```diff
@@ -0,0 +1,231 @@
+/**
+  Provides a consistent target interface, minimizing boilerplate and
+  automatically safeguarding declarations related to disabled options.
+
+  # Type
+
+  ```
+  mkTarget :: AttrSet -> ModuleBody
+  ```
+
+  Where `ModuleBody` is a module that doesn't take any arguments. This allows
+  the caller to use module arguments.
+
+  # Examples
+
+  The `modules/«MODULE»/«PLATFORM».nix` modules should use this function as
+  follows:
+
+  ```nix
+  { mkTarget, lib... }:
+  mkTarget {
+    name = "«name»";
+    humanName = "«human readable name»";
+
+    generalConfig =
+      lib.mkIf complexCondition {
+        home.packages = [ pkgs.hello ];
+      };
+
+    configElements = [
+      { programs.«name».theme.name = "stylix"; }
+
+      (
+        { colors }:
+        {
+          programs.«name».theme.background = colors.base00;
+        }
+      )
+
+      (
+        { fonts }:
+        {
+          programs.«name».font.name = fonts.monospace.name;
+        }
+      )
+    ];
+  }
+  ```
+
+  # Inputs
+
+  `config` (Attribute set)
```

Comment:
I assume this is the convention for notating that the function receives only one argument of type attribute set. And then this argument receives some arbitrary indicative name, like `config`.